catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java

```diff
@@ -808,7 +820,42 @@ public ResourceResponse retrieveResource(URI resourceUri,
             Map<String, Serializable> requestProperties)
             throws IOException, ResourceNotFoundException, ResourceNotSupportedException {
 
-        LOGGER.debug("retrieving resource at : {}", resourceUri);
+        if (resourceReader == null) {
+            // If no resource reader was found, retrieve the product through a GetRecordById request
+            Serializable serializableId = requestProperties.get(Metacard.ID);
+
+            if (serializableId == null) {
+                throw new ResourceNotFoundException(
+                        "Unable to retrieve resource because no metacard ID was found.");
+            } else {
```

Comment:
You don't need the else block anymore. Without removing it, you've not reduced your complexity. :)